processing/src/main/java/io/druid/segment/data/GenericIndexedWriter.java

```diff
@@ -81,7 +154,17 @@ public void write(T objectToWrite) throws IOException
     valuesOut.write(Ints.toByteArray(bytesToWrite.length));
     valuesOut.write(bytesToWrite);
 
-    headerOut.write(Ints.toByteArray((int) valuesOut.getCount()));
+    if (!requireMultipleFiles) {
+      writeIntValueToOutputStream(buf, (int) valuesOut.getCount(), headerOut);
```

Comment:
Not sure why github is not showing this line as `outdated`.
https://github.com/druid-io/druid/pull/3743/commits/19b69f335ded01a8069d84c06ab3f0afeeb1a90d#diff-37ba90c2e6d6b833ce38fc1ce76eecc3R158